assets/src/components/map.tsx

```diff
@@ -227,88 +229,66 @@ const EventAdder = ({
-    })
-  }
+export type UpdateMapFromPointsFn = (map: LeafletMap, points: LatLng[]) => void
+
+export interface FollowerProps {
+  positions: LatLng[]
+  onUpdate: UpdateMapFromPointsFn
 }
 
-const AutoCenter = ({
-  latLngs,
-  shouldAutoCenter,
-  isAutoCentering,
-}: {
-  shouldAutoCenter: boolean
-  isAutoCentering: MutableRefObject<boolean>
-  latLngs: LatLng[]
-}) => {
+export const Follower = ({
+  positions,
+  isUpdating,
+  onUpdate,
+  shouldFollow = true,
+}: FollowerProps & InteractiveFollowState) => {
   const map = useMap()
-
-  const [{ pickerContainerIsVisible }] = useContext(StateDispatchContext)
-  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(latLngs)
+  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(positions)
 
   if (
-    !equalByElements(latLngs, currentLatLngs, (latLng1, latLng2) =>
-      latLng1.equals(latLng2)
-    )
+    !equalByElements(positions, currentLatLngs, (lhs, rhs) => lhs.equals(rhs))
   ) {
-    setCurrentLatLngs(latLngs)
+    setCurrentLatLngs(positions)
   }
 
   useEffect(() => {
-    if (map !== null && shouldAutoCenter) {
-      isAutoCentering.current = true
-      autoCenter(map, currentLatLngs, pickerContainerIsVisible)
+    if (map !== null && shouldFollow) {
+      if (isUpdating !== undefined) {
+        isUpdating.current = true
+      }
+      onUpdate(map, currentLatLngs)
     }
-  }, [
-    map,
-    shouldAutoCenter,
-    isAutoCentering,
-    currentLatLngs,
-    pickerContainerIsVisible,
-  ])
+  }, [map, shouldFollow, isUpdating, currentLatLngs, onUpdate])
 
   return <></>
 }
+export interface InteractiveFollowState {
+  isUpdating: MutableRefObject<boolean>
+  shouldFollow: boolean
+  setShouldFollow: Dispatch<SetStateAction<boolean>>
+}
 
-const useAutoCenterState = () => {
-  const [shouldAutoCenter, setShouldAutoCenter] = useState<boolean>(true)
-  const isAutoCentering: MutableRefObject<boolean> = useRef(false)
+// Gathers all state needed for the Follower to be able to display it's state
+// as well as support turning off when interrupted
+export const useInteractiveFollowerState = (): InteractiveFollowState => {
+  const [shouldFollow, setShouldFollow] = useState<boolean>(true)
+  const isFollowing: MutableRefObject<boolean> = useRef(false)
 
   return {
-    shouldAutoCenter,
-    setShouldAutoCenter,
-    isAutoCentering,
+    shouldFollow,
+    setShouldFollow,
+    isUpdating: isFollowing,
   }
 }
 
-interface AutoCenterMapOnProps {
-  latLngs: LatLng[]
-  isAutoCentering: MutableRefObject<boolean>
-  shouldAutoCenter: boolean
-  setShouldAutoCenter: Dispatch<SetStateAction<boolean>>
-}
-
-const AutoCenterMapOn = ({
-  shouldAutoCenter,
-  setShouldAutoCenter,
-  isAutoCentering,
-  latLngs,
-}: AutoCenterMapOnProps) => {
+// Sets up map events to get a callback when the user interacts with the map
+// which should override the follower on state
+function useStopFollowingOnInteraction(
+  isAutoCentering: React.MutableRefObject<boolean>,
```

Comment:
question: should this be `isFollowing` or `isUpdating` for consistency?

Reply:
Yep! Thank you!

Reply:
https://github.com/mbta/skate/pull/1891#discussion_r1085875987